pkg/kubectl/cmd/util/printing.go

```diff
@@ -133,6 +134,25 @@ func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error
 	return maybeWrapSortingPrinter(cmd, printer), generic, nil
 }
 
+// PrintResourceInfoForCommand receives a *cobra.Command and a *resource.Info and
+// attempts to print an info object based on the specified output format. If the
+// object passed is non-generic, it attempts to print the object using a HumanReadablePrinter.
+// Requires that printer flags have been added to cmd (see AddPrinterFlags).
+func PrintResourceInfoForCommand(cmd *cobra.Command, info *resource.Info, f *Factory, out io.Writer) error {
+	printer, generic, err := PrinterForCommand(cmd)
+	if err != nil {
+		return err
+	}
+	if !generic || printer == nil {
+		printer, err := f.PrinterForMapping(cmd, nil, false)
+		if err != nil {
+			return err
+		}
+		return printer.PrintObj(info.Object, out)
```

Comment:
This line is using the printer declared in line [142](https://github.com/kubernetes/kubernetes/pull/38112/files/66f31212ac1d9cd8bc1b6cc5ea3c22b87bcaef0b#diff-6207a4b98b40755e352878f799b7677fR142) although I realize that this entire part could read a bit better. I'll go ahead and rename the `printer` declared inside of the `!generic || printer == nil` block to something a bit more specific, like `nonGenericPrinter`